app/controllers/applications_controller.rb

```diff
@@ -19,14 +19,26 @@ def update
     if @application.update(application_params) && params[:commit] == 'Apply Changes'
       redirect_to event_application_path(@event.id, @application.id),
       notice: "You have successfully updated your application for #{@event.name}."
-    elsif @application.update(application_params) && params[:commit] == 'Save Changes'
-      redirect_to event_application_path(@event.id, @application.id),
-      notice: "You have successfully saved your changes to the draft."
+    elsif params[:commit] == 'Save Changes'
+      @application.skip_validation = true
+      if @application.update(application_params)
+        redirect_to event_application_path(@event.id, @application.id),
+        notice: "You have successfully saved your changes to the draft."
+      end
     else
       render :edit
     end
   end
 
+  def submit
+    if @application.update(application_params)
+      @application.update_attributes(submitted: true)
```

Comment:
If the user is submitting the application from the draft form, we are not sending them the confirmation email that they have applied.